drivers/remote/driver.go

```
@@ -9,59 +11,163 @@ import (
 			log.Errorf("Error registering Driver for %s due to %v", name, err)
 		}
 	})
 	return nil
 }
 
-func (d *driver) Config(option map[string]interface{}) error {
-	return driverapi.ErrNotImplemented
+func (d *driver) call(methodName string, arg interface{}, retVal interface{}) error {
+	method := driverapi.NetworkPluginEndpointType + "." + methodName
+	return d.endpoint.Call(method, arg, retVal)
 }
 
-func (d *driver) CreateNetwork(id types.UUID, option map[string]interface{}) error {
+// Config is not implemented for remote drivers, since it is assumed
+// to be supplied to the remote process out-of-band (e.g., as command
+// line arguments).
+func (d *driver) Config(options map[string]interface{}) error {
 	return driverapi.ErrNotImplemented
 }
 
+func (d *driver) CreateNetwork(id types.UUID, options map[string]interface{}) error {
+	create := &createNetworkRequest{
+		NetworkID: string(id),
+		Options:   options,
+	}
+	return d.call("CreateNetwork", create, &createNetworkResponse{})
+}
+
 func (d *driver) DeleteNetwork(nid types.UUID) error {
-	return driverapi.ErrNotImplemented
+	delete := &deleteNetworkRequest{NetworkID: string(nid)}
+	return d.call("DeleteNetwork", delete, &deleteNetworkResponse{})
 }
 
 func (d *driver) CreateEndpoint(nid, eid types.UUID, epInfo driverapi.EndpointInfo, epOptions map[string]interface{}) error {
-	return driverapi.ErrNotImplemented
+	if len(epInfo.Interfaces()) > 0 {
+		return errors.New("Expected no interfaces in EndpointInfo")
+	}
+
+	create := &createEndpointRequest{
+		NetworkID:  string(nid),
+		EndpointID: string(eid),
+		Options:    epOptions,
+	}
+	var res createEndpointResponse
+	if err := d.call("CreateEndpoint", create, &res); err != nil {
+		return err
+	}
+
+	ifaces, err := res.parseInterfaces()
+	if err != nil {
+		return err
+	}
+	for _, iface := range ifaces {
+		if err := epInfo.AddInterface(iface.ID, iface.MacAddress, *iface.Address, *iface.AddressIPv6); err != nil {
```

Should we report the failure in calling `epInfo.AddInterface` back to the remote driver ?
Infact I think regardless of failure or success of the `epInfo.AddInterface`, it would be good to make an additional RPC call to make sure the remote Drivers have the exact same behavior as the in-built driver.